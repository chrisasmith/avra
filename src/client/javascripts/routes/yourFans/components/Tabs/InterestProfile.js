import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { injectIntl, defineMessages } from 'react-intl'
import styles from '../styles/YourFans.scss'
import * as yourFansActions from '../../../../reducers/pages/yourFansActions'
import { createLoadingSelector } from '../../../../reducers/selectors'
import DashboardGraphModule, { MODULE_TYPE } from '../../../../components/GridCards/DashboardGraphModule'
import FilterBreadCrumb from '../../../../components/Filter/FilterBreadCrumb'

const propTypes = {
  data: PropTypes.object,
  filter: PropTypes.object,
  isFetching: PropTypes.bool.isRequired,
  intl: PropTypes.object,
}

const messages = defineMessages({
  areasTitle: {
    id: 'your_fans.tab.interest.areas_title',
    defaultMessage: 'Interest Areas',
  },
  areasSideTitle: {
    id: 'your_fans.tab.interest.areas_side',
    defaultMessage: 'Interest areas',
  },
  additionalInterest: {
    id: 'your_fans.tab.interest.additional',
    defaultMessage: 'Additional Interests',
  },
  sportsInterestArea: {
    id: 'your_fans.tab.interest.sports_area',
    defaultMessage: 'Sports Interest Areas',
  },
})

const InterestProfileTab = injectIntl(({ intl, filter, data, isFetching }) => (
  <div className={styles.grid}>
    <div className={styles.breadCrumb}>
      <FilterBreadCrumb filter={filter} />
    </div>
    <div className="container">
      <div className="row row-grid">
        <div className="col-6">
          <DashboardGraphModule
            title={intl.formatMessage(messages.areasTitle)}
            sideTitle={intl.formatMessage(messages.areasSideTitle)}
            data={data ? data.interest_areas : null}
            isFetching={isFetching}
            type={MODULE_TYPE.PERCENT}
            propertyTitle={filter.property.name}
            benchmarkTitle={filter.benchmark.name}
          />
          <DashboardGraphModule
            title={intl.formatMessage(messages.additionalInterest)}
            sideTitle={intl.formatMessage(messages.areasSideTitle)}
            data={data ? data.additional_interests : null}
            isFetching={isFetching}
            type={MODULE_TYPE.PERCENT}
            propertyTitle={filter.property.name}
            benchmarkTitle={filter.benchmark.name}
          />
        </div>
        <div className="col-6">
          <DashboardGraphModule
            title={intl.formatMessage(messages.sportsInterestArea)}
            sideTitle={intl.formatMessage(messages.areasSideTitle)}
            data={data ? data.sports_interest_areas : null}
            isFetching={isFetching}
            type={MODULE_TYPE.PERCENT}
            propertyTitle={filter.property.name}
            benchmarkTitle={filter.benchmark.name}
          />
        </div>
      </div>
    </div>
  </div>
))

function mapStateToProps(state) {
  const loadingSelector = createLoadingSelector([yourFansActions.GET_HOME_PAGE.SELF])
  const filter = state.filters.filter
  const data = state.pages.yourFans.data
  return { ...state, isFetching: loadingSelector(state), filter, data }
}

InterestProfileTab.propTypes = propTypes
export default connect(mapStateToProps)(InterestProfileTab)
