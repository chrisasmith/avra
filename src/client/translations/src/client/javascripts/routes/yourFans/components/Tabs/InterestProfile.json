[
  {
    "id": "your_fans.tab.interest.areas_title",
    "defaultMessage": "Interest Areas"
  },
  {
    "id": "your_fans.tab.interest.areas_side",
    "defaultMessage": "Interest areas"
  },
  {
    "id": "your_fans.tab.interest.additional",
    "defaultMessage": "Additional Interests"
  },
  {
    "id": "your_fans.tab.interest.sports_area",
    "defaultMessage": "Sports Interest Areas"
  }
]