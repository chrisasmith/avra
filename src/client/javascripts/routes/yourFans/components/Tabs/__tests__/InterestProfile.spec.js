import React from 'react'
import { shallow } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { shallowToJson } from 'enzyme-to-json'
import InterestProfile from '../InterestProfile'

const mockStore = configureMockStore([thunk])

let store

const getTab = s => shallow(<InterestProfile store={mockStore(s)} />).shallow()

describe('Interest Profile Tab', () => {
  beforeEach(() => {
    store = {
      location: { },
      loading: {

      },
      error: {

      },
      filters: {
        filter: {
          property: {
            name: 'property',
          },
          benchmark: {
            name: 'benchmark',
          },
        },
      },
      pages: {
        yourFans: {
          data: null,
        },
      },
    }
  })

  it('should show', () => {
    const wrapper = getTab(store)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should show loading', () => {
    store.loading.GET_HOME_PAGE = true

    const wrapper = getTab(store)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
