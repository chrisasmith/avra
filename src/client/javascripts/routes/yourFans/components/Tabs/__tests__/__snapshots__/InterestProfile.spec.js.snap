// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interest Profile Tab should show 1`] = `
<div
  className="grid"
>
  <div
    className="breadCrumb"
  >
    <FilterBreadCrumb
      filter={
        Object {
          "benchmark": Object {
            "name": "benchmark",
          },
          "property": Object {
            "name": "property",
          },
        }
      }
    />
  </div>
  <div
    className="container"
  >
    <div
      className="row row-grid"
    >
      <div
        className="col-6"
      >
        <DashboardGraphModule
          benchmarkTitle="benchmark"
          data={null}
          isFetching={false}
          propertyTitle="property"
          sideTitle="Interest areas"
          title="Interest Areas"
          type="PERCENT"
        />
        <DashboardGraphModule
          benchmarkTitle="benchmark"
          data={null}
          isFetching={false}
          propertyTitle="property"
          sideTitle="Interest areas"
          title="Additional Interests"
          type="PERCENT"
        />
      </div>
      <div
        className="col-6"
      >
        <DashboardGraphModule
          benchmarkTitle="benchmark"
          data={null}
          isFetching={false}
          propertyTitle="property"
          sideTitle="Interest areas"
          title="Sports Interest Areas"
          type="PERCENT"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Interest Profile Tab should show loading 1`] = `
<div
  className="grid"
>
  <div
    className="breadCrumb"
  >
    <FilterBreadCrumb
      filter={
        Object {
          "benchmark": Object {
            "name": "benchmark",
          },
          "property": Object {
            "name": "property",
          },
        }
      }
    />
  </div>
  <div
    className="container"
  >
    <div
      className="row row-grid"
    >
      <div
        className="col-6"
      >
        <DashboardGraphModule
          benchmarkTitle="benchmark"
          data={null}
          isFetching={true}
          propertyTitle="property"
          sideTitle="Interest areas"
          title="Interest Areas"
          type="PERCENT"
        />
        <DashboardGraphModule
          benchmarkTitle="benchmark"
          data={null}
          isFetching={true}
          propertyTitle="property"
          sideTitle="Interest areas"
          title="Additional Interests"
          type="PERCENT"
        />
      </div>
      <div
        className="col-6"
      >
        <DashboardGraphModule
          benchmarkTitle="benchmark"
          data={null}
          isFetching={true}
          propertyTitle="property"
          sideTitle="Interest areas"
          title="Sports Interest Areas"
          type="PERCENT"
        />
      </div>
    </div>
  </div>
</div>
`;
